apollo/src/datasources/GitHubAPI.js

```diff
@@ -29,22 +17,89 @@ class GitHubAPI extends GraphQLDataSource {
     if (!request.headers) {
       request.headers = {};
     }
-
     request.headers.Authorization = `${this.token}`;
   }
 
+  /**
+   * Returns a new Error with a more helpful Error message.
+   *
+   * If Github throws an error because it cannot find a repo
+   * with the given ownder/name/until parameters, this function
+   * will let the client know that their parameters may be
+   * incorrect.
+   *
+   * @param {Error}     err        The Error Object to modify
+   * @param {String[]}  paramsArr  The params the user should have given
+   *
+   * @returns Error
+   */
+  helpfulErrorReturn(err, paramsArr) {
+    // This match statement can change in the future
+    //     depending on the errors thrown by the Github API
+    if (err.message.match(/(cannot read).+(defaultBranchRef)/i)) {
+      // Maps through our params to create a readable list
+      const paramsStr = paramsArr
+        .map((param, i) => {
+          const newParam = `"${param}"`;
+          // ↑ adds "quote" around each param
+          // ↓ if it's the last item, adds the word 'and'
+          if (i === paramsArr.length - 1) return 'and ' + newParam;
+          return newParam;
+        })
+        .join(', ');
+      err.message +=
+        '. Are the ' + paramsStr + ' properties typed correctly in this query?';
+    }
+    return err;
+  }
```

Comment:
Previously, if you had a query witha  typo like
```
SparkyBoy(owner: "DevWarr", name: "typohere-challenge") {
    oid
  }
```

Apollo qould not say _what_ could have caused the error.
This method checks if the Github error could be the result of params the user passed in, and then lets the user know to check their params 👍